src/main/java/com/lunark/lunark/controller/PropertyController.java

```diff
@@ -84,26 +91,62 @@ public ResponseEntity<Collection<PropertyAvailabilityEntry>> changePricesAndAvai
     public ResponseEntity<PropertyResponseDto> createProperty(@RequestBody PropertyRequestDto propertyDto) {
-        // TODO: add service calls
-        PropertyResponseDto response = modelMapper.map(propertyDto, PropertyResponseDto.class);
-        response.setAmenities(Arrays.asList(new PropertyAmenityDto("Wi-Fi", null), new PropertyAmenityDto("Washing machine", null)));
+        Property property = propertyService.create(modelMapper.map(propertyDto, Property.class));
+        PropertyResponseDto response = modelMapper.map(property, PropertyResponseDto.class);
         return new ResponseEntity<>(response, HttpStatus.CREATED);
     }
 
-    @PutMapping(consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
-    public ResponseEntity<PropertyResponseDto> updateProperty(@RequestBody PropertyRequestDto propertyDto) {
-        // TODO: add service calls
-        return new ResponseEntity<>(new PropertyResponseDto(), HttpStatus.OK);
+    @PutMapping(value = "/{id}", consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<PropertyResponseDto> updateProperty(@PathVariable("id") Long id, @RequestBody PropertyRequestDto propertyDto) {
+        Property property = modelMapper.map(propertyDto, Property.class);
+        property.setId(id);
+        property = propertyService.update(property);
+        PropertyResponseDto response = modelMapper.map(property, PropertyResponseDto.class);
+        return new ResponseEntity<>(response, HttpStatus.OK);
     }
 
     @DeleteMapping(value = "/{id}")
-    public ResponseEntity<PropertyResponseDto> deleteProperty(@PathVariable("id") Long id) {
-        // TODO: add service calls
+    public ResponseEntity<?> deleteProperty(@PathVariable("id") Long id) {
+        propertyService.delete(id);
         return new ResponseEntity<>(HttpStatus.NO_CONTENT);
     }
 
     @PostMapping(value = "/approve/{id}")
     public ResponseEntity<?> approveProperty(@PathVariable("id") Long id) {
         return new ResponseEntity<>(HttpStatus.NO_CONTENT);
     }
+
+    @GetMapping(value = "/{propertyId}/images/{imageId}", produces = MediaType.IMAGE_PNG_VALUE)
+    public ResponseEntity<byte[]> getImage(@PathVariable("propertyId") Long propertyId, @PathVariable("imageId") Long imageId) {
+        Optional<Property> property = propertyService.find(propertyId);
+        if (property.isEmpty()) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+
+        Optional<PropertyImage> image = propertyService.getImage(imageId, propertyId);
+        if (image.isEmpty()) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+
+        return ResponseEntity
+                .ok()
+                .contentType(MediaType.parseMediaType(image.get().getMimeType()))
+                .body(image.get().getImageData());
+    }
+
+    @PostMapping(value = "/{id}/images")
```

Comment:
When I tried to upload images Spring was telling me I was not sending multipart requests. This fixes the error.
```suggestion
    @PostMapping(value = "/{id}/images", consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
```